jdaviz/configs/default/plugins/markers/markers.py

```diff
@@ -144,56 +199,134 @@ def _recompute_mark_positions(self, viewer):
+                        continue
+
+                    wcs = self.app.data_collection[data_label].coords
+                    try:
+                        original_indices_mask = (qtable['data_label'] == data_label) & in_viewer
+
+                        ra_values = np.asarray(qtable['world_ra'][original_indices_mask]) * u.deg
+                        dec_values = np.asarray(qtable['world_dec'][original_indices_mask]) * u.deg
+
+                        new_x[original_indices_mask], new_y[original_indices_mask] = \
+                            wcs.world_to_pixel_values(ra_values, dec_values)
+
+                    except Exception:
+                        # fail gracefully
+                        new_x, new_y = [], []
+                        break
+
+            else:
+                raise NotImplementedError(f"align_by {self.app._align_by} not implemented")
+
+            viewer_mark.x, viewer_mark.y = new_x, new_y
+
+    def _add_distance_row(self, p1, p2):
+        """
+        Adds a single row to the measurements table with the results
+        of a distance measurement.
+        """
+        world_avail = ('world_ra' in p1 and 'world_ra' in p2 and
+                       p1.get('world_ra') is not None and p2.get('world_ra') is not None and
+                       np.all(np.isfinite([p1['world_ra'], p1['world_dec'],
+                                           p2['world_ra'], p2['world_dec']])))
+
+        if world_avail:
+            c1 = SkyCoord(p1['world_ra'], p1['world_dec'], unit='deg', frame='icrs')
+            c2 = SkyCoord(p2['world_ra'], p2['world_dec'], unit='deg', frame='icrs')
+            sep_arcsec = c1.separation(c2).arcsec
+            pos_angle = c1.position_angle(c2).deg
+            start_ra, start_dec = p1['world_ra'], p1['world_dec']
+            end_ra, end_dec = p2['world_ra'], p2['world_dec']
         else:
-            raise NotImplementedError(f"align_by {self.app._align_by} not implemented")
+            sep_arcsec, pos_angle, start_ra, start_dec, end_ra, end_dec = (np.nan,) * 6
+
+        dist_pix = np.sqrt((p2.get('pixel_x', 0) - p1.get('pixel_x', 0))**2 +
+                           (p2.get('pixel_y', 0) - p1.get('pixel_y', 0))**2)
+
+        self.distances_table.add_item({
+            'Start RA': f"{start_ra:.4f}" if not np.isnan(start_ra) else "N/A",
+            'Start Dec': f"{start_dec:.4f}" if not np.isnan(start_dec) else "N/A",
+            'End RA': f"{end_ra:.4f}" if not np.isnan(end_ra) else "N/A",
+            'End Dec': f"{end_dec:.4f}" if not np.isnan(end_dec) else "N/A",
+            'Separation (arcsec)': f"{sep_arcsec:.4f}" if not np.isnan(sep_arcsec) else "N/A",
+            'Distance (pix)': f"{dist_pix:.2f}",
+            'Position Angle (deg)': f"{pos_angle:.2f}" if not np.isnan(pos_angle) else "N/A"
+        })
+
+    def _get_snap_coordinates(self, viewer):
+        """
+        Finds the permanent marker in the table closest to the cursor's
+        current position in the viewer.

```

Comment:
(nothing actionable now, just leaving thoughts for the future)  Once we have catalogs as data-collection entries, I think we'll want to snap to those scatter points by default (following the behavior of the coords-info logic), but I still see how it would be useful to be able to manually snap to these markers as well.  Eventually we'll probably then need some visual cue for how this snapping is done.